app/models/poll.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+class Poll < ActiveRecord::Base
+  has_many :options, dependent: :destroy
+
+  validates :title, presence: true
+
+  after_touch :ensure_status_is_correct
+
+  def make_ready
```

Comment:
Shouldn't this be a full-fledged [ActiveRecord::Enum](http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html) instead of a homegrown solution?